Differentiating landlord reviews

jesshan700 opened the pull request:
### Added to landlord reviews <!-- Required -->

<!-- Provide a general summary of your changes in the Title above -->
<!-- Itemize bug fixes, new features, and other changes -->
<!-- Feel free to break this into sub-sections, i.e. features, fixes, etc. -->
<!-- Some examples are shown below. -->

This pull request is the first step towards showing the difference between landlord reviews and apartment reviews on the landlord page. Checks the aptId field for each review to see if it is empty or null indicating a landlord review.

- [x] Text that shows the user which reviews are landlord reviews


### Test Plan <!-- Required -->

<!-- Provide screenshots or point out the additional unit tests -->
Testing Lambrou landlord vs apartment:
<img width="837" alt="Screen Shot 2023-10-09 at 3 15 29 PM" src="https://github.com/cornell-dti/cu-apts/assets/125935795/d8b71b06-0530-4a16-9bf6-00e5c663da86">


Testing Travis Hyde landlord vs apartment:
<img width="837" alt="Screen Shot 2023-10-09 at 3 18 27 PM" src="https://github.com/cornell-dti/cu-apts/assets/125935795/9cdda2de-4914-403d-a079-735d871e3f24">


### Notes <!-- Optional -->

<!--- List any important or subtle points, future considerations, or other items of note. -->
- Decided not to make the text a link because it could get confusing for the user if the link just directs them to the page that they are already on. 
- Could maybe implement some sort of icon to show the differentiation vs text!

### Breaking Changes  <!-- Optional -->

<!-- Keep items that apply: -->
- Database schema change (anything that changes Firestore collection structure)
- Other change that could cause problems (Detailed in notes)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cornell-dti/cu-apts?pullRequest=305) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cornell-dti/cu-apts?pullRequest=305) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cornell-dti/cu-apts?pullRequest=305) it.</sub>

mluo24 commented on frontend/src/components/Review/Review.tsx:
```diff
@@ -162,6 +180,15 @@ const ReviewComponent = ({
+                  {/* Checking to see if aptId of the review is null or empty, indicating that the review is a 
+                  landlord apartment - if landlord apartment is true, will add landlord review text */}
+                  <Typography className={landlordIndicator}>
+                    {review.aptId === '' || review.aptId == null
```
I think you can do `!review.aptId` instead.

Also, with what I was saying above, if you add `landlordName` as a prop, you can rewrite the code as such:

```
{landlordName && (
<Typography className={landlordIndicator}>
  'Landlord Review - ' + landlordName
</Typography>
)}
```

This also only renders the component if `landlordName` has an input.

mluo24 commented on frontend/src/components/Review/Review.tsx:
```diff
@@ -99,14 +108,23 @@ const ReviewComponent = ({
     setExpanded(!expanded);
   };
 
+  //Retreieving landlord data
+  useEffect(() => {
+    get<Landlord>(`/api/landlord/${review.landlordId}`, {
+      callback: setLandlord,
+    });
+  }, [review]);
+
```
I think this API call isn't actually necessary. The parent component already should have made this API call and acquired the data already (since it's the landlord page), and since you only need the name of the landlord, it would make sense to just pass the name of it as a prop to this component, especially since this component is used in other places where you don't need the landlord name.

I would suggest adding an optional prop to this component that passes in the landlord name, and in 188 replace the `landlord?.name` with this prop, and maybe change the condition so it relies on the landlord name being not null instead.

mluo24 reviewed (commented):
Nice work Jessica! You really made the indicator look nice, and it's clear which one is a landlord review. Maybe you can also add indication for what's an apartment review with the apartment name?

Your logic with your solution is great, but I left some comments for how to micro-optimize it.

jesshan700 commented:
Updated the differentiation between landlord and apartments so that apartment links with the specific apartment shows on the card rather than showing the landlord review. Changed the styling so that it matches with the admin page.

<img width="1284" alt="Screen Shot 2023-10-16 at 4 06 49 PM" src="https://github.com/cornell-dti/cu-apts/assets/125935795/32f19a0d-3474-49e4-9bb2-72e16f2e64bf">


Redirects user to the specific apartment of that card. 

<img width="1284" alt="Screen Shot 2023-10-16 at 4 07 38 PM" src="https://github.com/cornell-dti/cu-apts/assets/125935795/2f07c19a-0e9f-4992-b526-c8b58c1a3361">

jesshan700 commented:
<img width="834" alt="Screen Shot 2023-10-22 at 10 31 33 PM" src="https://github.com/cornell-dti/cu-apts/assets/125935795/6784f6f9-d3fd-4724-bfbf-c1d291af7c15">

Changed styling of properties indicator rather than red link as shown above, better indicator that the link represents the property. Underlined portion is a link to the respective apartment page.

ggsawatyanon reviewed (approved):
This looks good, Jessica! This is a really helpful feature for users to understand the UI of the landlord reviews page easier. I'm not sure the purpose of the addition of the install_nvm.s file though. Good job on this PR!

thuypham03 reviewed (approved):
Good job on adding a differentiation between landlord and apartment's review on landlord page. Now, it's easier for the user to follow and can easy redirect to the corresponding landlord page.
